### PR TITLE
ipatests: Test for ipahealthcheck DogtagCertsConnectivityCheck

### PR DESCRIPTION
**Pull request by menonsudhir**
This test checks that when pki-tomcat service is stopped, DogtagCertsConnectivityCheck displays the result as ERROR and vice-versa.

### CODE REVIEW

**Comment by rcritten** [comment]
I wonder if this test exposes a bug.

The test depends in dirsrv running which I guess I did because it usually indicates that IPA is basically up.

It could instead require pki_tomcatd so it skips the test altogether if dogtag isn't running (really, what's the point) It would return:

`Source 'ipahealthcheck.dogtag.ca' is missing one or more requirements 'pki_tomcatd'
`
@fcami what do you think?

The purpose of the bug is to discover when the CA is running but something blew up such that it isn't responding to requests.

So how we could test that I don't know, I don't know of a way to provoke the CA to be running and not responding. Perhaps leaving it as-is so we can confirm that the check can execute is sufficient.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -305,6 +305,38 @@ def test_source_ipahealthcheck_dogtag_ca_dogtagcertsconfigcheck(self):
...
+        self.master.run_command(
+            ["systemctl", "start", "pki-tomcatd@pki-tomcat"]
+        )
+
+    def test_ipahealthcheck_dogtag_ca_connectivity_check(self, restart_tomcat):
+        """
+        This testcase checks that when the pki-tomcat service is stopped,
+        DogtagCertsConnectivityCheck displays the result as ERROR
+        and check displays the result as SUCCESS when pki-tomcat
+        service is running on IPA Master.
```
Comment (flo-renaud) [inline ipatests/test_integration/test_ipahealthcheck.py:324]: The comment is out-of-date (the test does not check SUCCESS).

**Comment by flo-renaud** [comment]
Hi @menonsudhir 
the test LGTM, except for my minor inline comment. I would like to ask for a temp commit with a test run but unfortunately PR-CI issue [363](https://github.com/freeipa/freeipa-pr-ci/issues/363) prevents successful execution in our infra.
However I was able to run the test on my VM. If others agree (@wladich @kaleemsiddiqu), I can ACK.
My run:
```
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd119] RUN ['systemctl', 'stop', 'pki-tomcatd@pki-tomcat']
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd119] Exit code: 0
[ipatests.pytest_ipa.integration.host.Host.vm-234.IPAOpenSSHTransport] RUN ['ipa-healthcheck', '--source', 'ipahealthcheck.dogtag.ca', '--check', 'DogtagCertsConnectivityCheck', '--output-type', 'json']
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120] RUN ['ipa-healthcheck', '--source', 'ipahealthcheck.dogtag.ca', '--check', 'DogtagCertsConnectivityCheck', '--output-type', 'json']
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120] ra.get_certificate(): Unable to communicate with CMS (503)
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120] [
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]   {
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "source": "ipahealthcheck.dogtag.ca",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "check": "DogtagCertsConnectivityCheck",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "result": "ERROR",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "uuid": "16be47f8-c85a-4134-9554-4887756f7cde",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "when": "20200528082620Z",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "duration": "0.041436",
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     "kw": {
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]       "msg": "Request for certificate failed, Certificate operation cannot be completed: Unable to communicate with CMS (503)"
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]     }
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120]   }
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120] ]
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd120] Exit code: 1
.[ipatests.pytest_ipa.integration.host.Host.vm-234.IPAOpenSSHTransport] RUN ['systemctl', 'start', 'pki-tomcatd@pki-tomcat']
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd121] RUN ['systemctl', 'start', 'pki-tomcatd@pki-tomcat']
[ipatests.pytest_ipa.integration.host.Host.vm-234.cmd121] Exit code: 0
```

**Comment by wladich** [comment]
Hi @flo-renaud, @menonsudhir: I do not see any issues besides the docstring.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 1 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: The doctsring has been updated.

**Comment by rcritten** [comment]
The dnf change was reverted in PR-CI, I think that a temp commit will work now.

**Comment by flo-renaud** [comment]
Temp commit shows the test is successful: [logs](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/dcd56b20-a4b8-11ea-ac0d-fa163e3ad6c5/report.html)
and the code LGTM

**Comment by rcritten** [comment]
master:

* 6a7fa03f91955cc999b9ae143647071fc35e3793 ipatests: Test for ipahealthcheck DogtagCertsConnectivityCheck

**Pull request closed without merging**